src/services/ehr-repo/__tests__/get-fragments.test.js

```diff
@@ -0,0 +1,141 @@
+import nock from 'nock';
+import { logError, logInfo } from '../../../middleware/logging';
+import { getAllFragmentsWithMessageIdsFromRepo } from "../get-fragments";
+import { downloadFromUrl } from "../../transfer/transfer-out-util";
+import { getMessageFragmentStatusByMessageId} from "../../database/message-fragment-repository";
+import { EhrUrlNotFoundError, errorMessages, PatientRecordNotFoundError } from "../../../errors/errors";
+
+jest.mock('../../../middleware/logging');
+jest.mock('../../../config', () => ({
+  config: jest.fn().mockReturnValue({
+    ehrRepoAuthKeys: 'fake-keys',
+    ehrRepoServiceUrl: 'http://fake-ehr-repo-url',
+    sequelize: { dialect: 'postgres' },
+    use_rds_credentials: false
+  })
+}));
+jest.mock('../../database/fragments-trace-repository');
+jest.mock('../../transfer/transfer-out-util');
+
+
+describe('getAllFragmentsFromRepo', () => {
+  describe('new ehr repo api', () => {
+
+    const mockEhrRepoAuthKeys = 'fake-keys';
+    const mockEhrRepoServiceUrl = 'http://fake-ehr-repo-url';
+
+    const nhsNumber = '1234567890';
+    const conversationIdFromEhrIn = 'fake-conversation-id'
+    const messageIds = ['fake-messageId1', 'fake-messageId2', 'fake-messageId3'];
+    const headers = {
+      reqheaders: { Authorization: `${mockEhrRepoAuthKeys}`}
+    };
+
+    const fragmentPresignedUrlRoot = 'http://fake-presigned-url/';
+
+    const ehrFragment = {
+      payload: "<?xml a very large xml>",
+      attachments: ["attachment1", "attachment2"],
+      "external_attachments": ["ext_attachment1", "ext_attachment2"]
+    };
+
+    it('should return the stored ehr fragments from ehr repo', async () => {
+      getMessageFragmentStatusByMessageId.mockReturnValue(null);
+
+      const repoScopeForIds = nock(mockEhrRepoServiceUrl, headers)
+        .get(`/patients/${nhsNumber}`)
+        .reply(200, {
+          "conversationIdFromEhrIn": conversationIdFromEhrIn,
+          "fragmentMessageIds": messageIds
+        });
+
+      const repoScopeForFragments = nock(mockEhrRepoServiceUrl, headers);
+      for (const messageId of messageIds) {
+        repoScopeForFragments
+          // TODO: currently this endpoint of ehr-repo doesn't work as expect. we might need to change this endpoint
+
+          .get(`/messages/${conversationIdFromEhrIn}/${messageId}`)
+          .reply(200, fragmentPresignedUrlRoot + messageId)

```

Comment:
```suggestion
        repoScopeForFragments
          // TODO: currently this endpoint of ehr-repo doesn't work as expect. we might need to change this endpoint

          .get(`/messages/${conversationIdFromEhrIn}/${messageId}`)
          .reply(200, fragmentPresignedUrlRoot + messageId)

```
To be addressed as part of a separate ticket?